Sources/Anthropic/Public/Parameters/Message/MessageParameter.swift

```diff
@@ -156,21 +161,127 @@ public struct MessageParameter: Encodable {
+            self.name = name
+            self.description = description
+            self.parameters = parameters
+        }
+        
+        public struct Parameter {
+            let name: String
+            let type: ParamType
+            let description: String
+            
+            public init(name: String, type: ParamType, description: String) {
+                self.name = name
+                self.type = type
+                self.description = description
+            }
+            
+            public enum ParamType: String {
+                case string
+                case integer
+                case number
+            }
+
+            public func toXML() -> String {
+                return """
+            <parameter>
+                <name>\(name)</name>
+                <type>\(type)</type>
+                <description>\(description)</description>
+            </parameter>
+            """
+            }
+        }
+        
+        public func toXML() -> String {
+            return """
+        <tool_description>
+            <tool_name>\(name)</tool_name>
+            <description>\(description)</description>
+            <parameters>
+                \(parameters.map { $0.toXML() }.joined(separator:"\n\t\t"))
+            </parameters>
+        </tool_description>
+        """
+        }
+    }
+    
+    enum CodingKeys: String, CodingKey {
+        case model
+        case messages
+        case maxTokens
+        case system
+        case metadata
+        case stopSequences
+        case stream
+        case temperature
+        case topK
+        case topP
+    }
+    
+    public func encode(to encoder: any Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(model, forKey: .model)
+        try container.encode(messages, forKey: .messages)
+        try container.encode(maxTokens, forKey: .maxTokens)
+        var systemStr = system ?? ""
+        if functions.count > 0 {
+            systemStr += toolsPreamble
+            systemStr += functions.compactMap { $0.toXML() }.joined(separator: "\n")
+        }
+        try container.encode(systemStr, forKey: .system)
+        try container.encode(metadata, forKey: .metadata)
+        try container.encode(stopSequences, forKey:  .stopSequences)
+        try container.encode(stream, forKey: .stream)
+        try container.encode(temperature, forKey: .temperature)
+        try container.encode(topK, forKey: .topK)
+        try container.encode(topP, forKey: .topP)
+    }
+    
+    private let toolsPreamble = """
+In this environment you have access to a set of tools you can use to answer the user's question.
+
+You may call them like this:
+<function_calls>
+<invoke>
+<tool_name>$TOOL_NAME</tool_name>
+<parameters>
+<$PARAMETER_NAME>$PARAMETER_VALUE</$PARAMETER_NAME>
+...
+</parameters>
+</invoke>
+</function_calls>
+
+Here are the tools available:
+"""
```

Comment:
i'd link the doc this comes from so its clear its the expressly-suggested way to integrate this https://docs.anthropic.com/claude/docs/functions-external-tools